odxtools/cli/_print_utils.py

```diff
@@ -53,31 +63,197 @@ def print_diagnostic_service(
+        if not getattr(service.request, "required_parameters", None):
+            if len(service().hex().upper()) > 24:
+                request.append('0x' + service().hex().upper()[:24] + '...')
+            else:
+                request.append('0x' + service().hex().upper())
+        else:
+            request.append(None)
+
+    return {'Name': name, 'Semantic': semantic, 'Hex-Request': request}
+
+
+def extract_parameter_tabulation_data(parameters: List[Parameter]) -> Dict[str, Any]:
+    # extracts data of parameters of diagnostic services into Dictionary which can be printed by tabulate module
+
+    name = []
+    byte = []
+    bit_length: List[Optional[int]] = []
+    semantic = []
+    param_type = []
+    value: List[Optional[str]] = []
+    value_type: List[Optional[str]] = []
+    data_type: List[Optional[str]] = []
+    dop: List[Optional[str]] = []
+
+    for param in parameters:
+        assert isinstance(param, Parameter)
+        name.append(param.short_name)
+        byte.append(param.byte_position)
+        semantic.append(param.semantic)
+        param_type.append(param.parameter_type)
+        if param.get_static_bit_length() is not None:
+            bit_length.append(param.get_static_bit_length())
+            length = (param.get_static_bit_length() or 0) // 4
+        else:
+            bit_length.append(None)
+        if isinstance(param, CodedConstParameter):
+            if isinstance(param.coded_value, int):
+                value.append("0x" + f"{param.coded_value:0{length}X}")
+            else:
+                value.append(f"{param.coded_value!r}")
+            data_type.append(param.diag_coded_type.base_data_type.name)
+            value_type.append('coded value')
+            dop.append(None)
+        elif isinstance(param, NrcConstParameter):
+            data_type.append(param.diag_coded_type.base_data_type.name)
+            value.append(str(param.coded_values))
+            value_type.append('coded values')
+            dop.append(None)
+        elif isinstance(param, (PhysicalConstantParameter, SystemParameter, ValueParameter)):
+            dop.append(param.dop.short_name)
+            if (tmp := getattr(param, "physical_type", None)) is not None:
+                data_type.append(tmp.base_data_type.name)
+            else:
+                data_type.append(None)
+            if isinstance(param, PhysicalConstantParameter):
+                if isinstance(param.physical_constant_value, int):
+                    value.append("0x" + f"{int(param.physical_constant_value):0{length}X}")
+                else:
+                    value.append(f"{param.physical_constant_value!r}")
+                value_type.append('constant value')
+            elif isinstance(param, ValueParameter) and param.physical_default_value is not None:
+                if isinstance(param.physical_default_value, int):
+                    value.append("0x" + f"{int(param.physical_default_value):0{length}X}")
```

Comment:
Why force all ints to be hex?

Reply:
also, better use
```suggestion
                    value.append(f"0x{int(param.physical_default_value):0{length}X}")
```
instead of concatinating strings...

Reply:
> Why force all ints to be hex?

Doesn't is make sense to display the values as hex as they're sent this way? I don't mind displaying ints as ints if it makes more sense though.

Reply:
the values are sent as binary, not hex, hex is just a human interpretation of numeric values and binary data.

it would not make sense to show a variable that represents a physical quantity (time, speed, distance, temperature) as hex.

Reply:
as a rule of thumb, if it has a unit, hex makes no sense.